actions/docker-publish/action.yaml

```diff
@@ -1,60 +1,61 @@
     required: true
   password:
     description: "Password for the Docker registry login."
     required: true
   working-directory:
-    description: "Working directory for your Docker artifacts. (Default is .)"
+    description: "Path of the directory containing the Dockerfile."
+    required: true
+  docker-registry:
+    description: "Host where the image should be pushed to."
+    required: true
+  github-token:
+    description: "Github token."
+    required: true
+  ref:
+    description: "Ref to checkout"
     required: false
-    default: "."
+    default: ${{ github.event.repository.default_branch }}
 runs:
   using: "composite"
   steps:
-    - name: Download Docker image tar artifact
-      uses: actions/download-artifact@v3
-      with:
-        name: ${{ inputs.image-artifact-name }}
-        path: build
-
-    - name: Login to DockerHub
-      uses: docker/login-action@v1
+    - name: Check out repository
+      uses: actions/checkout@v3
       with:
-        username: ${{ inputs.username }}
-        password: ${{ inputs.password }}
+        ref: ${{ inputs.ref }}
+        token: ${{ inputs.github-token }}
 
-    - name: Publish Docker image
-      run: |
-        if [[ $(ls -1 build/*.tar 2>/dev/null | wc -l) != 1 ]]; then
-          >&2 echo "Error: A single image tar file is needed in the downloaded artifact. You can upload one before using this action: https://github.com/actions/upload-artifact."
-          exit 1
-        fi
+    # setup-buildx action will create and boot a builder using by default the docker-container driver.
+    # This is not required but recommended using it to be able to build multi-platform images, export cache, etc.
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2.2.1
 
-        docker load --input build/*.tar
+    - name: Login to the Registry
+      uses: "docker/login-action@v2.1.0"
+      with:
+        registry: "${{ inputs.docker-registry }}"
+        username: "${{ inputs.username }}"
+        password: "${{ inputs.password }}"
 
-        if [[ "$GITHUB_REF" =~ ^refs/tags/.* ]]; then
-          docker tag ${{ inputs.image-name }}:${{ github.run_id }} ${{ inputs.publisher }}/${{ inputs.image-name }}:latest
-          docker tag ${{ inputs.image-name }}:${{ github.run_id }} ${{ inputs.publisher }}/${{ inputs.image-name }}:${GITHUB_REF/refs\/tags\//}
-          docker push ${{ inputs.publisher }}/${{ inputs.image-name }}:latest
-          docker push ${{ inputs.publisher }}/${{ inputs.image-name }}:${GITHUB_REF/refs\/tags\//}
-        else
-          docker tag ${{ inputs.image-name }}:${{ github.run_id }} ${{ inputs.publisher }}/${{ inputs.image-name }}:${{ github.run_id }}
-          docker push ${{ inputs.publisher }}/${{ inputs.image-name }}:${{ github.run_id }}
-        fi
-      shell: bash
-      working-directory: ${{ inputs.working-directory }}
+    - name: Build and push
+      uses: docker/build-push-action@v3.2.0
```

Comment:
What about publishing a Docker image that was build with [Jib](https://github.com/GoogleContainerTools/jib) from Java source code? This action should only publish Docker images because we need to support all use cases and building a Docker image from a Dockerfile can be provided as a workflow template if needed

Reply:
This was also the reason for allowing a `.tar` file as an input